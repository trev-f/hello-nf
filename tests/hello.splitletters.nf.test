nextflow_process {

    name "Test Process splitLetters"
    script "hello.nf"
    process "splitLetters"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = file("test-file.txt")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("set params manually") {

        when {
            params {
                outdir = "output"
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = file("test-file.txt")
                """
            }
        }

        then {
            assert params.outdir == "output"
        }

    }

    test("set nested params manually") {

        when {
            params {
                output {
                    dir = "output"
                }
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = file("test-file.txt")
                """
            }
        }

        then {
            assert params.output.dir == "output"
        }

    }

    test("load params from yaml") {

        when {
            params {
                load("${baseDir}/tests/params.yaml")
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = file("test-file.txt")
                """
            }
        }

        then {
            assert params.outdir == "output"
        }

    }

    test("load nested params from yaml") {

        when {
            params {
                load("${baseDir}/tests/params_nested.yaml")
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = file("test-file.txt")
                """
            }
        }

        then {
            assert params.output.dir == "output"
        }

    }

}
