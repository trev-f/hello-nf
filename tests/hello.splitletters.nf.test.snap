{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "chunk_aa:md5,d1a7fb5eab1c16cb4f7cf341cf188c3d",
                        "chunk_ab:md5,08cf82251c975a5e9734699fadf5e9c0"
                    ]
                ]
            }
        ],
        "timestamp": "2023-09-15T13:27:10+0000"
    }
}